README.md

```diff
@@ -22,21 +21,50 @@ might require you to write more application code.
 
 Learn more at [pekko.apache.org](https://pekko.apache.org/docs/pekko-http/current/).
 
-Documentation
--------------
+## Documentation
 
 The documentation is available at
 [pekko.apache.org](https://pekko.apache.org/docs/pekko-http/current/), for
 [Scala](https://pekko.apache.org/docs/pekko-http/current/scala/http/) and
 [Java](https://pekko.apache.org/docs/pekko-http/current/java/http/).
 
-
-Community
----------
+## Building from Source
+
+### Prerequisites
+- Make sure you have installed a Java Development Kit (JDK) version 8 or later.
```

Comment:
I woukd sayJDK 8 only (i.e. remove the later part) because Pekko Http build needs to produce JDK 8 bytecode. If you build with after JDK then JDK 8 users cannot run Pekko Http.


This can be resolved with the `-release` flag which pekko http uses but with conditionals which may not be relevant now (i.e. latest versions of Scala 2 fixes some problems) but the point is we need to be careful about this and verify it.

Reply:
our pekko repo says JDK 11+

Reply:
The critical one, i.e. building jar artifacts is JDK 8 and we inherited that from Akka https://github.com/apache/incubator-pekko-http/blob/main/.github/workflows/publish.yml#L19 .

Also a distinction should be made between building release artifacts to validate release process and just working on the code locally.

I am on my phone atm so it's a bit hard to read entire thing in context but for me the important part is that it's clear that when creating release artifacts (for verification or doing the release yourself as a release manager) that you use JDK 8.

And it might well be that it cresting release artifacts for JDK 11 or higher is fine but after going through the process of validating/improving/clarifying this in root pekko i would rather be safe

Reply:
this appears in our build.sbt

```
    (if (isJdk8) Seq.empty
     else if (scalaBinaryVersion.value == "2.12") Seq("-target:jvm-1.8")
     else Seq("-release", "8")),
```

Reply:
Yes I know but this was for older versions of Scala. `-target` in the latest versions of Scala is either deprecated or removed and ontop of the the behaviour of `-release` has changed.

Reply:
I built pekko-http on my laptop with JDK11 and was then able to use the jars that I published to MavenLocal with JDK 8 builds of downstream projects. The classes appear to be Java 8 format.

Reply:
I am more concerned about stuff like unsafe or missing references to documentation. I'll check this properly when I get home

Reply:
The safe option is to build with JDK 8. The compatibility mode of JDK 11 will probably work most of the times but we ran into issues in the past, where the JDK signature storage that allows JDK 11 to figure out which methods are present in JDK 8 was not completely accurate.